extensions/amp-lightbox-gallery/0.1/amp-lightbox-gallery.js

```diff
@@ -758,16 +765,43 @@ export class AmpLightboxGallery extends AMP.BaseElement {
    * @private
    */
   shouldExit_() {
-    const element = this.getCurrentElement_().sourceElement;
-    return this.shouldAnimate_(element)
+    const target = this.getCurrentElement_().sourceElement;
+    if (!this.transitionTargetIsInViewport_(target)) {
+      return false;
+    }
+    return this.shouldAnimate_(target)
         .then(shouldAnimate => {
-          const transitionBackToSource = element == this.sourceElement_;
-          const belongsToCarousel =
-            this.manager_.hasCarousel(this.currentLightboxGroupId_);
-          return shouldAnimate && (transitionBackToSource || belongsToCarousel);
+          return shouldAnimate;
         });
   }
 
+  /**
+   *
+   * @param {!Element} target
+   * @returns {boolean}
+   * @private
+   */
+  transitionTargetIsInViewport_(target) {
```

Comment:
This method feels a little weird where 3/4 paths have no side effects and then 1 path has a side effect.

Reply:
Yeah this deserves a bit of a refactor.